bika/lims/browser/analysisrequest/add.py

```diff
@@ -475,11 +477,12 @@ def __call__(self):
         self.context.plone_utils.addPortalMessage(message, 'info')
         # Automatic label printing won't print "register" labels for Secondary. ARs
         new_ars = [ar for ar in ARs if ar[-2:] == '01']
+        new_ar_uids = [uid_dict[ar_id] for ar_id in new_ars]
         if 'register' in self.context.bika_setup.getAutoPrintStickers() \
-                and new_ars:
+                and new_ar_uids:
```

Comment:
I'm just being so much overprecise here. To speed up boolean evaluations, Python uses short-circuit evaluations. It means that boolean evaluation may stop if one of its expression is False. In this case, is faster if you rephrase the condition like follows:

```python
if new_ar_uids and 'register' in self.context.bika_setup.getAutoPrintStickers():
```